### PR TITLE
fixed the allure report dir

### PR DESCRIPTION
**Pull request by PeterNgTr**

### CODE REVIEW

File: lib/config.js [hunk 1]
```diff
@@ -14,6 +14,9 @@ const defaultConfig = {
   hooks: [],
   gherkin: {},
   plugins: {
+    allure: {
+      outputDir: './_output'
```
[the remark this answers is not in the document]
Comment (PeterNgTr) [reply lib/config.js]: added trailing comma

File: lib/codecept.js [hunk 2]
```diff
@@ -46,6 +46,7 @@ class Codecept {
   initGlobals(dir) {
     global.codecept_dir = dir;
     global.output_dir = fsPath.resolve(dir, this.config.output);
+    global.allure_dir = fsPath.resolve(dir, this.config.plugins.allure.outputDir);
```
Comment (DavertMik) [inline lib/codecept.js]: please don't add this global

File: lib/command/run.js [hunk 3]
```diff
@@ -25,11 +26,19 @@ module.exports = function (test, options) {
...
+  else allureDir = path.join(testRoot, config.plugins.allure.outputDir);
+
   if (!fileExists(outputDir)) {
     output.print(`creating output directory: ${outputDir}`);
     mkdirp.sync(outputDir);
   }
 
+  if (!fileExists(allureDir)) {
+    output.print(`creating output directory: ${allureDir}`);
+    mkdirp.sync(allureDir);
```
Comment (DavertMik) [inline lib/command/run.js]: this should be moved into plugin itself

File: lib/config.js [hunk 4]
```diff
@@ -14,6 +14,9 @@ const defaultConfig = {
   hooks: [],
   gherkin: {},
   plugins: {
+    allure: {
+      outputDir: './_output',
```
Comment (DavertMik) [inline lib/config.js]: also... this should be moved to plugin itself. THere is `defaultConfig` variable, where you can set defaults

File: lib/codecept.js (hunk 2 above)
Comment (PeterNgTr) [reply lib/codecept.js]: removed this.

File: lib/config.js (hunk 4 above)
Comment (PeterNgTr) [reply lib/config.js]: @DavertMik could you help with this, I'm not really sure about this, if I remove this on config.js, the tests are failed.

Comment (DavertMik) [reply lib/config.js]: Yeah, inside plugin function you receive the `config` variable. What you can do there is to merge default values with those received in config:

```js
const defaultConfig = {
  outputDir: './output',
};

module.exports = (config) => {
  config = Object.assign(defaultConfig, config);
   //...
}
```

File: lib/plugin/allure.js [hunk 5]
```diff
@@ -1,5 +1,10 @@
 const event = require('../event');
 const Allure = require('allure-js-commons');
+const mkdirp = require('mkdirp');
+const output = require('../output');
+const path = require('path');
+const fileExists = require('../utils').fileExists;
```
[the remark this answers is not in the document]
Comment (DavertMik) [reply lib/plugin/allure.js]: Please run 
```
npm run lint-fix
```
to fix all such errors

File: lib/config.js (hunk 4 above)
Comment (PeterNgTr) [reply lib/config.js]: It didn't help. Those tests are failed when I removed those 2 lines in lib/config.js
`
    allure: {
      outputDir: './_output',
`

`
 BDD Gherkin
    1) should run feature files
    2) should print substeps in debug mode
    3) should run feature with examples files
    4) should run feature with tables
    5) should run feature with long strings
    6) should run feature by file name
    7) should run feature by scenario name
    8) should run feature by tag name
    9) should run scenario by tag name
    ✓ should show all available steps (319ms)
    ✓ should generate snippets for missing steps (339ms)

  CodeceptJS Runner
    10) should be executed in current dir
    11) should be executed with glob
    12) should be executed with config path
    13) should show failures and exit with 1 on fail
    14) should run bootstrap
    15) should run teardown
    16) should run async bootstrap
    17) should run hooks
    18) should run hooks from suites
    19) should run hooks from suites (in different order)
    20) should run different types of scenario
    21) should run bootstrap/teardown as object
    22) should run dynamic config
    23) should run dynamic config with profile
    24) should run dynamic config with profile 2

  Codeceptjs Events
    25) should fire events with only passing tests
    26) should fire events with passing and failing tests

  CodeceptJS Interface
    27) should rerun flaky tests
    28) should rerun retried steps
    29) should include grep option tests
    30) should run tests with different data
    31) should execute expected promise chain
    32) should display meta steps and substeps

  list/def commands
    ✓ list should print actions (338ms)
    ✓ def should create definition file (379ms)
    ✓ def should create definition file given a config file (413ms)

  CodeceptJS Multiple Runner
    33) should execute one suite with browser
    34) should execute all suites
    35) should replace parameters
    36) should execute multiple suites
    37) should execute multiple suites with selected browsers
    38) should print steps
    39) should pass grep to configuration
    40) should pass tests to configuration
    41) should run chunks

  CodeceptJS session
    42) should run with 3 sessions
    43) should run session defined before executing
    44) should run all session tests

  CodeceptJS within
    45) should execute if no generators
    46) should execute with generators. Yield is first in order
    47) should execute with generators. Yield is second in order
    48) should execute with async/await. Await is first in order
    49) should execute with async/await. Await is second in order
    50) should complete test steps after within
    51) should stop test execution after fail in within
    52) should stop test execution after fail in main block
    53) should return correct result after tests


  5 passing (24s)
  53 failing
`

**Comment by PeterNgTr** [comment]
this is fixed by another PR.

**Pull request closed without merging**